XMonad/Layout/ConditionalLayout.hs

```diff
@@ -0,0 +1,191 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StrictData #-}
```

Comment:
Is this really necessary? I generally regard `StrictData` as cargo-culting instead of understanding laziness.

Reply:
I mean, I don't see any reason *not* to have either of

``` haskell
data CondModifiedLayout c m l a = CondModifiedLayout Bool c (ModifiedLayout m l a)
data CondChoose         c l r a = CondChoose         Bool c (Choose l r a)
```

with strict fields. If you want, I can also add explicit annotations instead

Reply:
I'm just deeply suspicious of `StrictData`. You can talk to `c_wraith` in `#haskell` for a good rundown of why it's generally not a good idea.

Reply:
There is also that one now has to check three different places to determine strictness: strictness annotations, `StrictData` at the top of the file, `StrictData` in `default-extensions` of the Cabal file. I strongly prefer they be where they're used instead of acting at a distance.